geoindicators/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/RsuIndicatorsTests.groovy

```diff
@@ -399,4 +400,97 @@ class RsuIndicatorsTests {
         }
         assertEquals(0.177, concat)
     }
+
+
+    @Test
+    void surfaceFractionsTest() {
+        H2GIS h2GIS = H2GIS.open('./target/surface_fractions_db;AUTO_SERVER=TRUE')
+        h2GIS.load(SpatialUnitsTests.class.getResource("road_test.shp"), true)
+        h2GIS.load(SpatialUnitsTests.class.getResource("building_test.shp"), true)
+        h2GIS.load(SpatialUnitsTests.class.getResource("veget_test.shp"), true)
+        h2GIS.load(SpatialUnitsTests.class.getResource("hydro_test.shp"), true)
+        h2GIS.load(SpatialUnitsTests.class.getResource("zone_test.shp"),true)
+
+        def  prepareData = Geoindicators.SpatialUnits.prepareRSUData()
+        assertTrue prepareData.execute([zoneTable: 'zone_test', roadTable: 'road_test',  railTable: '',
+                                        vegetationTable : 'veget_test',
+                                        hydrographicTable :'hydro_test',
+                                        prefixName: "prepare_rsu", datasource: h2GIS])
+
+        def outputTableGeoms = prepareData.results.outputTableName
+
+        assertNotNull h2GIS.getTable(outputTableGeoms)
+
+        def rsu = Geoindicators.SpatialUnits.createRSU()
+        assertTrue rsu.execute([inputTableName: outputTableGeoms, prefixName: "rsu", datasource: h2GIS])
+        def outputTable = rsu.results.outputTableName
+
+        def  p =  Geoindicators.RsuIndicators.smallestCommunGeometry()
+        assertTrue p.execute([
+                              rsuTable: outputTable,buildingTable: "building_test", roadTable:"road_test",vegetationTable: "veget_test",waterTable: "hydro_test",
+                              prefixName: "test", datasource: h2GIS])
+        def outputTableStats = p.results.outputTableName
+
+        h2GIS.execute """DROP TABLE IF EXISTS stats_rsu;
+                    CREATE INDEX ON $outputTableStats (ID_RSU);
+                   CREATE TABLE stats_rsu AS SELECT b.the_geom,
+                round(sum(CASE WHEN a.low_vegetation=1 THEN a.area ELSE 0 END),1) AS low_VEGETATION_sum,
+                round(sum(CASE WHEN a.high_vegetation=1 THEN a.area ELSE 0 END),1) AS high_VEGETATION_sum,
+                round(sum(CASE WHEN a.water=1 THEN a.area ELSE 0 END),1) AS water_sum,
+                round(sum(CASE WHEN a.road=1 THEN a.area ELSE 0 END),1) AS road_sum,
+                round(sum(CASE WHEN a.building=1 THEN a.area ELSE 0 END),1) AS building_sum, a.id_rsu
+                FROM $outputTableStats AS a, $outputTable b WHERE a.id_rsu=b.id_rsu GROUP BY b.id_rsu;
```

Comment:
I am not quite sure I got the shape of the data you obtain before to do that. Actually I have in mind two possibilities representing the same thing:
- 1st possibility:
ID_RSU | ID_smallest_geo | low_veget | high_veget | water | road | building |
1            |        1                   |        1        |          1       |  0       |    0   |    0         |
1            |        2                   |        0        |          1       |  1       |    0   |    0         |

- 2nd possibility:
ID_RSU | ID_smallest_geo | low_veget | high_veget | water | road | building |
1            |        1                   |        1        |          0       |  0       |    0   |    0         |
1            |        1                   |        0        |          1       |  0       |    0   |    0         |
1            |        2                   |        0        |          1       |  0       |    0   |    0         |
1            |        2                   |        0        |          0       |  1       |    0   |    0         |

Can you turn on the light ? =)